components/scream/src/interface/scream_scorpio_interface.F90

```diff
@@ -257,15 +257,15 @@ subroutine eam_init_pio_2()
     ! Register all dimensions with the output file
     call register_dimension("example_pio_structured_v2.nc","x","horizontal distance",10)
     call register_dimension("example_pio_structured_v2.nc","y","vertical distance",3)
-    call register_dimension("example_pio_structured_v2.nc","t","time",0)
+    call register_dimension("example_pio_structured_v2.nc","time","time",0)
     ! Register all variables with the output file
-    call register_variable("example_pio_structured_v2.nc","t","time",1,(/ "t" /), PIO_double,"t")
+    call register_variable("example_pio_structured_v2.nc","time","time",1,(/ "time" /), PIO_double,"t")
     call register_variable("example_pio_structured_v2.nc","x","answer to space and time",1,(/ "x" /), PIO_double,"x-real")
     call register_variable("example_pio_structured_v2.nc","y","answer to space and time",1,(/ "y" /), PIO_double,"y-real")
-    call register_variable("example_pio_structured_v2.nc","bar","answer to space and time",3,(/ "x", "y", "t" /), PIO_double,"xyt-real")
-    call register_variable("example_pio_structured_v2.nc","foo","answer to space and time",3,(/ "x", "y", "t" /), PIO_double,"xyt-real")
-    call register_variable("example_pio_structured_v2.nc","bar_flip","answer to space and time",3,(/ "y", "x", "t" /), PIO_double,"yxt-real")
-    call register_variable("example_pio_structured_v2.nc","foo_flip","answer to space and time",3,(/ "y", "x", "t" /), PIO_double,"yxt-real")
+    call register_variable("example_pio_structured_v2.nc","bar","answer to space and time",3,(/ "x", "y", "time" /), PIO_double,"xyt-real")
```

Comment:
I think the right approach would be the latter you suggested.  I don't think we want to force all dimensions to have the same number of characters.  So what I would need to do is create an array of characters that is allocatable, allocate it to be the correct size, populate the array, pass it, then deallocate?

As a followup, in phase 2 I should be calling `register_variable` from C++ not Fortran.  Is there a GCC sensitivity to how that should be handled if shortnames are different lengths?

Reply:
I mean, this is simply a test, so you can create the arrays in whatever way you want. You are "forcing" same dims only in this small test.

As for F-C interoperability, strings are quite delicate (C puts a `\0` at the end (so it can find out length), wihle F doesn't (cause it attaches some descriptors to the string itself)). I am not sure how an array of strings would work, you might have to google it. I suspect it should not be too hard, but you might need to check what the almighty google has to say about it.

To be clear: the different length issue is only related to how you create the array (with `(/ "a", "abc" /)`). Being on the fly, it is basically equivalent to a `parameter` variable, which I _think_ is what causes the issue. When passing an array of strings, in general, there should not be any issue. In fact, all the gfortran errors you get are at the _call_ place, not inside the routine. The routine itself is fine with all strings in the array having different lengths.